main/process-tx.zkasm

```diff
@@ -68,9 +68,34 @@ endCheckChainId:
                                         :CALL(isColdAddress) ; add tx.origin to touched addresses
         0                               :MSTORE(depth) ; Initial depth is 0
 
-;; Set gasPrice global var
+;; Set gasPrice global var depending on effectivePercentage [0-255] -> txGasPrice = Floor((gasPrice * (effectivePercentage + 1)) / 256)
+        ; _effGasPriceShifted = gasPrice * (effectivePercentage + 1)
+        ; A => gasPrice
         $ => A                          :MLOAD(txGasPriceRLP)
-        A                               :MSTORE(txGasPrice)
+        ; B => effectivePercentage
+        $ => B                          :MLOAD(effectivePercentageRLP)
+        ; B => [0, 255]
+        B + 1 => B
+        ; A*B + C = D * 2**256 + op(E)
+        0 => C
+        ; _effGasPriceShifted = gasPrice * (effectivePercentage + 1)
+        $${var _effGasPriceShifted = A * B}
+        ; get value above 256 bits
+        ${_effGasPriceShifted >> 256} => D
+        ; compute ARITH
+        ${_effGasPriceShifted} => E              :ARITH
```

Comment:
Should we have out of arith counters check for this code?
(or any other counters (steps, binary?) given that this adds quite a bit of computation)

Reply:
Yes, I agree we should add an arith checker at beginning of `process-tx.zkasm`